src/main/java/com/commercetools/sync/commons/utils/StreamUtils.java

```diff
@@ -32,38 +27,29 @@
     }
 
     /**
-     * Resolves all not empty optionals of a given {@link Stream} and returns them as a {@link List}.
+     * Similar to Java 9
+     * <a href="https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#stream--">Optional#stream()</a>:
+     * converts Optional to a stream of one element if not empty. Otherwise - return an empty stream.
      *
-     * @param elements the stream of elements.
-     * @return a {@link List} of resolved optionals.
+     * @param optional Optional to convert
+     * @param <T>      type of optional value
+     * @return Stream of one element from the supplied {@code optional} if not empty, otherwise - an empty stream.
      */
     @Nonnull
-    public static <T extends List<?>> T asList(@Nonnull final Stream<? extends Optional> elements) {
-        return filterStream(elements, Collectors.toList());
+    public static <T> Stream<T> filterEmptyOptionals(final @Nonnull Optional<T> optional) {
```

Comment:
I would also put it in a new utils `OptionalUtils` not `StreamUtils` as its operating it's a util for `Optional` types

Reply:
> I would name the method toStream instead of filterEmptyOptionals as it only takes one optional and simply converts to a stream if it is present.

I tried: it doesn't seem so, for example:
``` 
 .flatMap(Utils::filterEmptyOptionals) // here i see explicitly from the name what's going on 
or
 .flatMap(Utils::toStream) // just some stream flatted to another stream
```

Reply:
> I would also put it in a new utils OptionalUtils not StreamUtils as its operating it's a util for Optional types

60d0a9b4

Reply:
@andrii-kovalenko-ct 
> // just some stream flatted to another stream

That's what exactly `OptionalUtils` would show explicitly when using `OptionalUtils::toStream`, you see exactly that an optional is being converted to a stream.

Reply:
but `OptionalUtils::toStream` doesn't looks verbose enough.
At least for me, despite that fact i implemented the method and know what it does. Guess other developers would be even more confused:
  - maybe this method converts stream to optional? 
  - or merges multiple optionals to common stream?
  - or converts nullable instances to stream of optionals?